xarray/backends/cfgrib_.py

```diff
@@ -25,6 +25,12 @@
 from ..core import indexing
 from ..core.utils import Frozen, FrozenOrderedDict
 from .common import AbstractDataStore, BackendArray
+from .file_manager import CachingFileManager
+from .locks import ensure_lock, SerializableLock
+
+# FIXME: Add a dedicated lock just in case, even if ecCodes is supposed to be thread-safe in most
+# circumstances. See: https://confluence.ecmwf.int/display/ECC/Frequently+Asked+Questions
+ECCODES_LOCK = SerializableLock()
 
 
 class CfGribArrayWrapper(BackendArray):
```

Comment:
This class will need a few updates (e.g., to make use of the `lock` from the datastore). Take a look at pynio for examples.

Reply:
Done.